powerplantmatching/osm/enhancement/reconstruction.py

```diff
@@ -0,0 +1,348 @@
+        self.similarity_threshold = config.get("units_reconstruction", {}).get(
+            "name_similarity_threshold", 0.7
+        )
+
+    def aggregate_names(self, names: set[str]) -> str:
+        """Aggregate multiple names into one representative name.
+
+        Parameters
+        ----------
+        names : set[str]
+            Generator names to aggregate
+
+        Returns
+        -------
+        str
+            Aggregated name or empty string
+        """
+        names_list = list(names)
+
+        if len(names_list) == 1:
+            return names_list[0]
+
+        if len(names_list) == 0:
+            return ""
+
+        common_name = self._find_common_substring(names_list)
+
+        if common_name:
+            return common_name
+
+        return names_list[0]
+
+    def _find_common_substring(self, names: list[str]) -> str | None:
+        """Find common substring among names using n-gram analysis."""
+        if not names:
+            return None
+
+        all_tokens = []
+        for name in names:
+            tokens = re.split(r"[\s\-_,\.]+", name)
+            all_tokens.append([t.lower() for t in tokens if t])
+
+        common_ngrams = {}
+
+        for n in range(1, 4):
+            for tokens in all_tokens:
+                for i in range(len(tokens) - n + 1):
+                    ngram = " ".join(tokens[i : i + n])
+                    if ngram not in common_ngrams:
+                        common_ngrams[ngram] = 0
+                    common_ngrams[ngram] += 1
+
+        threshold = len(names) * self.similarity_threshold
+        frequent_ngrams = {
+            ngram: count for ngram, count in common_ngrams.items() if count >= threshold
+        }
+
+        if not frequent_ngrams:
+            return None
+
+        longest_ngram = max(frequent_ngrams.keys(), key=lambda x: len(x))
+
+        for name in names:
+            if longest_ngram in name.lower():
+                start_idx = name.lower().index(longest_ngram)
+                return name[start_idx : start_idx + len(longest_ngram)].strip()
+
+        return longest_ngram.title()
+
+
+class PlantReconstructor:
+    """Reconstructs plants from orphaned generators.
+
+    Analyzes groups of generators within plant boundaries to reconstruct
+    missing plant data by aggregating generator attributes.
+
+    Attributes
+    ----------
+    config : dict
+        Reconstruction configuration
+    name_aggregator : NameAggregator
+        Name aggregation utility
+    generator_parser : GeneratorParser
+        Parser for processing generators
+    min_generators : int
+        Minimum generators needed for reconstruction
+
+    Examples
+    --------
+    >>> reconstructor = PlantReconstructor(config, name_aggregator)
+    >>> if reconstructor.can_reconstruct(len(generators)):
+    ...     plant_info = reconstructor.aggregate_generator_info(generators, country)
+    """
+
+    def __init__(
+        self,
+        config: dict[str, Any],
```

Comment:
This class only needs the `min_generators_for_reconstruction` - why give it the whole config?